fe/fe-core/src/main/java/com/starrocks/lake/StarMgrMetaSyncer.java

```diff
@@ -236,33 +236,53 @@ public int deleteUnusedWorker() {
         return cnt;
     }
 
-    @Override
-    protected void runAfterCatalogReady() {
-        deleteUnusedShardAndShardGroup();
-        deleteUnusedWorker();
-    }
+    public void syncTableMetaAndColocationInfo() {
+        List<Long> dbIds = GlobalStateMgr.getCurrentState().getDbIds();
+        for (Long dbId : dbIds) {
+            Database db = GlobalStateMgr.getCurrentState().getDb(dbId);
+            if (db == null) {
+                continue;
+            }
+            if (db.isSystemDatabase()) {
+                continue;
+            }
 
-    // do a one time sync, delete all shards from this table that exist in starmgr but not in fe
-    public static void syncTableMeta(String dbName, String tableName, boolean forceDeleteData) throws DdlException {
-        Database db = GlobalStateMgr.getCurrentState().getDb(dbName);
-        if (db == null) {
-            throw new DdlException(String.format("db %s does not exist.", dbName));
+            List<Table> tables = null;
+            Locker locker = new Locker();
+            locker.lockDatabase(db, LockType.READ);
+            try {
+                tables = db.getTables();
+            } finally {
+                locker.unLockDatabase(db, LockType.READ);
+            }
```

Comment:
```suggestion
            tables = db.getTables();
```
It should be fine to get table list without holding database lock.